### ISSUE TITLE
player character freeze when respawn

### ISSUE THREAD

**Issue by MilkissWhite** [qouteall/ImmersivePortalsModForForge#40]
MC 1.15.2
Forge 31.1.14 / 31.1.16 tested
Modlist
immersive-portals-0.4(for+mc+1.15+with+forge)
MixinBootstrap-1.0.2

---

 player dead and push respawn.
 player character respawn at spawnpoint. but freeze(can't move).
 I can see my arm but my character block my camera(I can see only inside of my skin).
 I can switch view to 3rd person camera by push F5.
 when I move my mouse, camera rotate but character do not.
 if my gamode is survival, mobs can interact me. for example, zombies can attack me. my character take damage but do not be knockback.
 if I open inventory, it will be close immediatly. advancement window function normally.
 finally, when I click anywhere(bare-hand attack), game crash with error message "attempting to attack an invalid entity". then go to multi-player server list screen even I was playing single-player game.
 I can continue to play if I just go to single-player mod again and just load world once more.

 it happen high chance at first death, but not always.
 while I testing, /kill command can repruduce but natural death with survival mode has more chance.
 after I changed forge version 31.1.16 from 13.1.14, it seems reproduce chance decreased but may be just feeling.
 I tested only without bed. and single-player. and not uninstalled resource packs when I testing.

 here are screenshots

01 - after respawn. this is inside of my character skin.
![MC_Bug001](https://user-images.githubusercontent.com/60180416/75245356-e7ce4f00-5810-11ea-931a-2d5dd37d32c9.png)

02 - after switch view. my character don't look at camera.
![MC_Bug002](https://user-images.githubusercontent.com/60180416/75245365-eac93f80-5810-11ea-94be-bbffc73feaec.png)

03 - error message after click
![MC_Bug003](https://user-images.githubusercontent.com/60180416/75245374-edc43000-5810-11ea-9458-8ed8c9788105.png)

and here is my latest log file.
[latest.log](https://github.com/qouteall/ImmersivePortalsModForForge/files/4249835/latest.log)

**Comment by qouteall** [comment]
Maybe there are two client player entities and it should be only one. Maybe the old player was not removed after respawn.

**Comment by MilkissWhite** [comment]
sorry, I can't understand. did I make "two client player entities" by doing something wrong in game-play?

**Comment by qouteall** [comment]
I tested forge 31.1.16 with only this mod installed and cannot reproduce this issue.

**Comment by qouteall** [comment]
I didn't mean that you are doing anything wrong.

**Comment by qouteall** [comment]
You installed some mods and uninstalled it but it will leave some registry terms in the world saving. But that should not cause any issue.

**Comment by qouteall** [comment]
werid

**Comment by qouteall** [comment]
I checked the code. "attack invalid entity" is caused by attacking yourself.

**Comment by MilkissWhite** [comment]
hmm.. when I found this issue first time, I have lot mods. but after then, I backup mods and try to find what mod cause this crash.
 finally I found crash doesn't happen when immersive portal removed. so I tested again with only portal and mixinbootstarap with new world. and it happen again, so I decide to report issue at this github.
 actually, when I tested last time for take screenshot, I changed my client language to english-us. and creative mode + /kill command doesn't cause crash. so I thought 'is this only korean language issue?' then I change to korean language, but still crash not happened. then I changed my gamemode to survival, and go to water. after that this issue happened again.
 you can see it screenshot 01, 02 is korean, but 03 is english. I changed language to english after character freeze, before bare-hand attack to air.
 so basically I forgot testing "english language + survival gamemode". I hope this information helps you.

**Comment by qouteall** [comment]
I tested korean language and couldn't reproduce this issue-_-

**Comment by MilkissWhite** [comment]
oh sorry, I have no clue anymore...

**Comment by qouteall** [comment]
I will make it check this kind of abnormality and manually fix it in the next version.
